horovod/run/elastic/registration.py

```diff
@@ -130,6 +135,13 @@ def _action(self):
     def _on_workers_recorded(self):
         logging.info('all {} workers recorded'.format(self.size()))
 
+        # Check that we have already reset the maximum number of allowed times
```

Comment:
I think we can reset once too often and then fail here even if all workers succeed if we check the limit here. We should check it right before we increment the counter further down, and then for `≥`.

Reply:
Yes, good point.  Fixed.